sound/soc/sof/topology.c

```diff
@@ -936,6 +952,13 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 		/* next array */
 		array = (struct snd_soc_tplg_vendor_array *)((u8 *)array
 			+ asize);
+
+		/* update parse position */
+		*read_bytes = *read_bytes + asize;
+
+		/* return when all tokens have been found */
+		if (found == count)
+			return 0;
```

Comment:
you could just add this to your loop condition as `while (priv_size > 0 && found != count)`

Reply:
@lyakh I have problem with this function, so should we return error value if not all tokens are found? I think we shouldn't have situation where some tokens are defined but some are not? then I could return here 0 and something like -EINVAL at the end...